dev-tool/src/list_net/data_source.rs

```diff
@@ -326,22 +333,26 @@ impl InMemorySamples {
         target_prob_dist: ArrayView1<f32>,
     ) -> Result<(), Error> {
         if inputs.shape() != [target_prob_dist.len(), ListNet::INPUT_NR_FEATURES] {
-            bail!("Sample with bad array shapes. Expected shapes [{nr_docs}, {nr_feats}] & [{nr_docs}] but got shapes {inputs_shape:?} & {prob_dist_shape:?}",
+            bail!("Sample with bad array shapes. Expected shapes [{nr_docs}, {nr_feats}] & [{nr_docs}] but got shapes {inputs_shape:?} & {prob_dist_shape:?}.",
                 nr_docs=inputs.shape()[0],
                 nr_feats=ListNet::INPUT_NR_FEATURES,
                 inputs_shape=inputs.shape(),
                 prob_dist_shape=target_prob_dist.shape(),
             );
         }
-        let mut data = Vec::with_capacity(inputs.len() + target_prob_dist.len());
-        extend_vec_with_ndarray(&mut data, inputs);
-        extend_vec_with_ndarray(&mut data, target_prob_dist);
-        self.data.push(data);
+        let mut datum = Vec::with_capacity(inputs.len() + target_prob_dist.len());
+        extend_vec_with_ndarray(&mut datum, inputs);
+        extend_vec_with_ndarray(&mut datum, target_prob_dist);
+        self.data.push(datum);
         Ok(())
     }
 }
 
-/// Extend a vec with the elements of given array in logical order.
+/// Extend a vec with the elements of the given array.
+///
+/// Elements are added in logical order independent of storage order,
+/// i.e. elements are added as if `data` was continuos and in standard
```

Comment:
*continuous

Reply:
:+1: Hm, why isn't `continuos` highlighted by the spell checker :thinking: Now the word is flagged :shrug: